resource_transip_vps.go

```diff
@@ -296,6 +300,29 @@ func resourceVpsRead(d *schema.ResourceData, m interface{}) error {
 	return nil
 }
 
+func resourceVpsUpdate(d *schema.ResourceData, m interface{}) error {
+	client := m.(repository.Client)
+	repository := vps.Repository{Client: client}
+
+	vps := vps.Vps{
+		Name:             d.Id(), // Unique ID provided by TransIP
+		Description:      d.Get("description").(string),
+		CPUs:             d.Get("cpus").(int),
```

Comment:
Why add CPUs, DiskSize and MemorySize to the update? They are computed properties.

Also if availability_zone is not passed, will it be reset to a default value or remain the same as the currently configured one?

Reply:
Good point, I based them on the example cURL calls from TransIP where they used all request attributes. Reading the documentation again I noticed only the following attributes can be updated:

- isCustomerLocked
- description
- tags

I'll set ForceNew for Description to `false`, remove unnecessary attributes and add the tags attribute to the update function.